src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
+type Shape = 'Triangle' | 'Circle' | 'Rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    // this.shape = 'triangle';
+    const checkSides = [a, b, c].sort((sideA, sideB) => (sideA - sideB));
 
-export class Circle {
+    if (checkSides.some((side) => (side <= 0))) {
+      throw new Error('Wrong triangle sides size, them < 0');
+    }
 
+    if (checkSides[2] >= checkSides[1] + checkSides[0]) {
+      throw new Error('longest side cannot be >= sum of 2 others');
+    }
+    // if (
+    //   this.a <= 0
+    //   || this.b <= 0
+    //   || this.c <= 0
+    //   || this.a + this.b <= this.c
+    //   || this.b + this.c <= this.a
+    //   || this.c + this.a <= this.b) {
```

Comment:
You could store these expressions as array elements
```
const conditions = [this.a <= 0, this.a + this.b <= this.c, and so on]
```
If one of them returns true, you should throw an error

Reply:
fixed it